pkg/logs/internal/decoder/single_line_handler_test.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-present Datadog, Inc.
+
+package decoder
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+func TestSingleLineHandlerProcess(t *testing.T) {
+	truncateTag := message.TruncatedReasonTag("single_line")
+	tagTrunLogsFlag := pkgconfigsetup.Datadog().GetBool("logs_config.tag_truncated_logs")
+	defer pkgconfigsetup.Datadog().SetWithoutSource("logs_config.tag_truncated_logs", tagTrunLogsFlag)
+
+	scenarios := []struct {
+		name             string
+		input            [][]byte
+		expected         [][]byte
+		expTags          [][]string
+		tagTruncatedLogs bool
+	}{
+		{
+			name:             "Base case",
+			input:            [][]byte{[]byte("This is a"), []byte("normal set  "), []byte(" of messages")},
```

Comment:
NIT: We can reduce the number of type annotations and type conversions by treating the inputs as strings, and then converting them to byte slices in the test body. 

IMO this increases readability of the test + makes refactoring slightly easier in the future. 
FWIW this is mostly my preference and not a rule of thumb in the agent code base, so feel free to disregard if you disagree